sqlglot/dialects/duckdb.py

```diff
@@ -88,6 +88,57 @@
 MAX_BIT_POSITION = exp.Literal.number(32768)
 
 
+def _to_boolean_sql(self: DuckDB.Generator, expression: exp.ToBoolean) -> str:
+    """
+    Transpile TO_BOOLEAN function from Snowflake to DuckDB equivalent.
+
+    DuckDB's CAST to BOOLEAN supports most of Snowflake's TO_BOOLEAN strings except 'on'/'off'.
+    We need to handle the 'on'/'off' cases explicitly, plus NaN/INF error cases.
+
+    In Snowflake, NaN and INF values cause errors. We use DuckDB's native ERROR()
+    function to replicate this behavior with a clear error message.
+    """
+    arg = expression.this
+
+    cast_to_real = exp.func("TRY_CAST", arg, exp.DataType.build("REAL"))
+
+    # Check for NaN and INF values
+    nan_inf_check = exp.Or(
+        this=exp.func("ISNAN", cast_to_real), expression=exp.func("ISINF", cast_to_real)
+    )
+
+    case_expr = exp.Case(
```

Comment:
Perhaps we can use the builder here to clean it up a bit? Check the docstring example:

https://github.com/tobymao/sqlglot/blob/c118af2f78af3e557f569c31b1561802338a48c4/sqlglot/expressions.py#L10103-L10121